### PR TITLE
Wire cutter - Allow use when unit has RHS Engineer UMBTS backpack

### PR DESCRIPTION
**Pull request by Dystopian**
**When merged this pull request will:**
- title.

~~Class name `rhs_assault_umbts_engineer` is hard-coded on purpose because I didn't find any analogues in vanilla, RHS and CUP gear.~~

### CODE REVIEW

**Comment by Blutze** [comment]
Gives it a 5 mass capacity advantage over standardised Assault Packs. Considering most of that is carried on the outside, I'd be ok with that.

Not sure what to think about the precedent of adding functionality to other mods simply based on cosmetic aspects of their models though. And how well does that code work when you don't have RHS loaded? Does it still work if it is replaced with some kind of macro in case other examples crop up?

File: addons/logistics_wirecutter/script_component.hpp [hunk 1]
```diff
@@ -56,3 +56,8 @@
 #define SOUND_CLIP_TIME_SPACING 1.5
 #define CUT_TIME_DEFAULT 11
 #define CUT_TIME_ENGINEER 7.5
+
+#define HAS_WIRECUTTER(unit) (\
+    "ACE_wirecutter" in ([ARR_6(unit, false, true, true, true, false)] call CBA_fnc_uniqueUnitItems) \
+    || {backpack unit isKindOf "rhs_assault_umbts_engineer"} \
```
Comment (dedmen) [inline addons/logistics_wirecutter/script_component.hpp]: Backpacks that inherit from this are not guaranteed to have a model with a wirecutter on it.

**Comment by Dystopian** [comment]
> And how well does that code work when you don't have RHS loaded?

It works OK. I thought about init checking if RHS is loaded and then replacing code with just `false` but I found it superfluous.
> Does it still work if it is replaced with some kind of macro in case other examples crop up?

If there are more examples it would be very easy to add their support. Until then there is no sense to do it.

**Comment by Blutze** [comment]
> I would implement it if there was at least very small probability for other such backpacks.

@jokoho48 mentioned VSM and another mod that even had them on vests. Haven't touched either in ages, but even if there isn't anything in those - I'd still try and keep this future proof.

File: addons/logistics_wirecutter/script_component.hpp [hunk 2]
```diff
@@ -59,5 +59,6 @@
 
 #define HAS_WIRECUTTER(unit) (\
     "ACE_wirecutter" in ([ARR_6(unit, false, true, true, true, false)] call CBA_fnc_uniqueUnitItems) \
-    || {backpack unit isKindOf "rhs_assault_umbts_engineer"} \
+    || {1 == getNumber (configFile >> "CfgVehicles" >> (backpack unit) >> QGVAR(hasWirecutter))} \
+    || {1 == getNumber (configFile >> "CfgWeapons" >> (vest unit) >> QGVAR(hasWirecutter))} \
```
Comment (dedmen) [inline addons/logistics_wirecutter/script_component.hpp]: Vests are CfgVehicles.

Comment (PabstMirror) [reply addons/logistics_wirecutter/script_component.hpp]: They are Items/CfgWeapons.
There is a `vestContainer` command, that returns the vest "object"
but `typeOf vestContainer player` = `"Supply140"`

Comment (Dystopian) [reply addons/logistics_wirecutter/script_component.hpp]: [Vests are CfgWeapons](https://community.bistudio.com/wiki/Arma_3_Characters_And_Gear_Encoding_Guide#Vest_configuration)
[Vest ground holders are CfgVehicles](https://community.bistudio.com/wiki/Arma_3_Characters_And_Gear_Encoding_Guide#Vest_ground_holder)

**Pull request merged**